packages/cli/src/executions/legacy-sqlite-execution-recovery.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Logger } from '@n8n/backend-common';
+import { GlobalConfig } from '@n8n/config';
+import { ExecutionData, ExecutionRepository } from '@n8n/db';
+import { Service } from '@n8n/di';
+
+/**
+ * Service for recovering executions that are missing execution data, this should only happen
+ * for sqlite legacy databases.
+ */
+@Service()
+export class LegacySqliteExecutionRecoveryService {
+	private readonly legacySqlite: boolean;
+
+	constructor(
+		private readonly logger: Logger,
+		private readonly executionRepository: ExecutionRepository,
+		private readonly globalConfig: GlobalConfig,
+	) {
```

Comment:
Can we set a log scope so all the debug logs can be shorter? See `this.logger.scoped`